addons/phone_validation/models/mail_thread_phone.py

```diff
@@ -47,6 +50,29 @@ class PhoneMixin(models.AbstractModel):
             when there is both a mobile and phone field in a model.")
     phone_mobile_search = fields.Char("Phone/Mobile", store=False, search='_search_phone_mobile_search')
 
+    def init(self):
+        super().init()
+        phone_fields = [
+            fname for fname in self._phone_get_number_fields()
+            if fname in self._fields and self._fields[fname].store
+        ]
+        # Add supporting indexes for searching on `phone_mobile_search`
+        for fname in phone_fields:
+            regex_expression = rf"regexp_replace(({fname}::text), '{PHONE_REGEX_PATTERN}'::text, ''::text, 'g'::text)"
+            # The btree index covers operators '=' and '=like' with a known prefix
+            create_index(self.env.cr,
+                         indexname=f'{self._table}_{fname}_partial_tgm',
+                         tablename=self._table,
+                         expressions=[regex_expression],
+                         where=f'{fname} IS NOT NULL')
+            # The trigram index covers operators 'like', 'ilike' and '=like' starting with a wildcard
+            create_index(self.env.cr,
```

Comment:
The trigram extension is not required.
This will fail for databases without it.
```suggestion
            if odoo.tools.db.has_trigram(self.env.cr):
                # The trigram index covers operators 'like', 'ilike' and '=like' starting with a wildcard
                create_index(self.env.cr,
```